backend/corpora/lambdas/api/v1/authentication.py

```diff
@@ -0,0 +1,82 @@
+from flask import make_response, jsonify, current_app, request, redirect
+import json
+from authlib.integrations.flask_client import OAuth
+from urllib.parse import urlencode
+from functools import lru_cache
+import base64
+from ....common.authorizer import get_userinfo, assert_authorized_token
+from ....common.corpora_config import CorporaAuthConfig
+
+
+@lru_cache(maxsize=1)
+def create_oauth_client(config):
+    oauth = OAuth(current_app)
```

Comment:
could this be instantiated in a more central location, to prevent creating multiple? Maybe it can be created in corpora_config and imported there.

Reply:
It's only created in this one file.   The lru_cache should make it so that only one is ever created.   Although I could have also implemented as a global variable in this module, which might have been clearer (there can be only one config).   Putting it in corpora_config would be fine, except then I'd have to import a few things into corpora_config (like flask.current_app, and OAuth), which doesn't seem right.  I wanted this file to be the only one that knew about authlib.